hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/DefaultS3ClientFactory.java

```diff
@@ -361,6 +366,13 @@ private static URI getS3Endpoint(String endpoint, final Configuration conf) {
    */
   private static Region getS3RegionFromEndpoint(String endpoint) {
 
+    // S3 VPC endpoint parsing
+    Matcher matcher = VPC_ENDPOINT_PATTERN.matcher(endpoint);
+    if(matcher.find()) {
+      LOG.debug("Endpoint {} is vpc endpoint; parsing", endpoint);
```

Comment:
so this is going to match on anything with .vpce. isn't it? I think it should include amazonaws.{com,com.cn} at the end so if someone ever sets up an internal host called vpce there's no confusion.

Reply:
Updated PR to match entire endpoint to avoid the confusion.